src/components/StoreList/StoreList.js

```diff
@@ -29,17 +17,49 @@ export default class StoreList extends React.Component {
 					items={this.state.storeList}
 					onClick={(storeCode, itemName) => {
 						try {
-							sendSlackMessage(
-								this.props.username,
-								itemName,
-								storeCode
+							let id = getUserSlackID(
+								this.props.currentUser,
+								this.props.users
 							);
+							if (!id) throw new Error();
+							sendSlackMessage(id, itemName, storeCode);
 						} catch (error) {
-							alert(error);
+							this.setState({sendSlackMessageError: true});
 						}
 					}}
 				/>
+				{this.state.sendSlackMessageError && (
+					<ErrorMessage text={'failed to send Slack message'} />
+				)}
 			</div>
 		);
 	}
 }
+
+const mapStateToProps = state => {
+	return {
+		storeList: state.storeList
+	};
+};
+
+const mapDispatchToProps = dispatch => {
+	return {
+		setStore: getStore((err, items) => {
```

Comment:
It'd be good to add redux thunk, this way our actions could return thunks that let them dispatch asynchronously and access the store if they wanted